### PR TITLE
Isaicp 2736

### PR DESCRIPTION
**Pull request by brummbar**
By submitting a pull request to the Joinup repository:
- you accept the conditions in the Joinup Individual Contributor Assignment Agreement
- you accept the terms laid out in the the code of conduct.

### CODE REVIEW

File: tests/features/custom_page/custom_page.search_field.feature [hunk 1]
```diff
@@ -0,0 +1,38 @@
+@api
+Feature: "Custom page" search.
+  In order to use the search
+  As a user of the website
+  I need to be able to view search results in the "Custom page" canonical page.
+
+  Background:
+    Given the following collection:
+      | title  | Code camp  |
```
Comment (pfrenssen) [inline tests/features/custom_page/custom_page.search_field.feature]: Don't reuse existing strings for new tests. "Code camp" belongs to the Add Custom Page feature.

File: tests/features/custom_page/custom_page.search_field.feature [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    And "custom_page" content:
+      | title    | collection | body               |
+      | About us | Code camp  | This is an example |
+    # Non UATable step.
+    Given I commit the solr index
+
+    Scenario: View search results in the "Custom page" canonical page.
+      When I am an anonymous user
+      And I go to the homepage of the "Code camp" collection
+      And I click "About us"
```
Comment (pfrenssen) [inline tests/features/custom_page/custom_page.search_field.feature]: There are two "About us" links on the page. One is in the navigation menu, and one is in a tile in the main content. Which one should I click? Is it intentional this is shown twice on the page?

This seems out of scope for this PR so I made a separate ticket for this: https://webgate.ec.europa.eu/CITnet/jira/browse/ISAICP-2780

File: web/modules/custom/joinup_search/joinup_search.module [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+use Drupal\Core\Access\AccessResult;
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\FieldItemListInterface;
+use Drupal\Core\Session\AccountInterface;
+
+/**
+ * Implements hook_entity_field_access().
+ *
+ * Grand access to fields of type search only to moderators.
```
Comment (pfrenssen) [inline web/modules/custom/joinup_search/joinup_search.module]: Typo: should be "grant".

File: tests/features/custom_page/custom_page.search_field.feature [hunk 4]
```diff
@@ -0,0 +1,38 @@
+@api
+Feature: "Custom page" search.
+  In order to use the search
+  As a user of the website
+  I need to be able to view search results in the "Custom page" canonical page.
```
Comment (pfrenssen) [inline tests/features/custom_page/custom_page.search_field.feature]: It is not the user of the website that wants this :D It's the collection owners that want to expose their content.

And nobody wants to view search results in the "Custom page" canonical page. We want to see search results in the search results page. For the end user this is just an overview of related content. They don't even know that this is built with a search engine in the backend. And they don't know what a canonical page is :)

File: tests/features/custom_page/custom_page.search_field.feature [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+      | title    | collection | body               |
+      | About us | Code camp  | This is an example |
+    # Non UATable step.
+    Given I commit the solr index
+
+    Scenario: View search results in the "Custom page" canonical page.
+      When I am an anonymous user
+      And I go to the homepage of the "Code camp" collection
+      And I click "About us"
+      Then I should see the heading "Discussion example"
```
Comment (pfrenssen) [inline tests/features/custom_page/custom_page.search_field.feature]: I don't see the heading "Discussion example" here. I have clicked on "About us" and obviously I see the heading "About us" now. This test is only green because the title of the "Discussion example" tile happens to be rendered with a H2 element which is also considered as a header.

The correct step would be:

> Then I should see the "Discussion example" tile in the related content section

The functionality is wrong, we cannot accept it like this. Why is the expected behaviour that when the user created an About Us page there are links to unrelated discussions being shown on the page? As a collection owner I want to have control over what appears on my custom pages. A simple custom page should not have any related content on it by default.

This test should actually be inverted. It should be like this:

```
Scenario: Related content does not show by default on a custom page
  Given I am an anonymous user
  When I go to the homepage of the "Code camp" collection
  And I click "About us"
  Then I SHOULD NOT SEE the "Discussion example" tile in the related content section
```

**Comment by brummbar** [comment]
Merged by mistake and reverted.
